src/System.Net.Sockets.Legacy/tests/FunctionalTests/DnsEndPointTest.cs

```diff
@@ -159,7 +159,7 @@ public void Socket_ConnectAsyncDnsEndPoint_Success()
             bool willRaiseEvent = sock.ConnectAsync(args);
             if (willRaiseEvent)
             {
-                complete.WaitOne();
+                Assert.True(complete.WaitOne(5000), "Timed out while waiting for connection");
```

Comment:
I see a lot of literal `5000` numbers.  Would a `const int TimeoutValueInMilliseconds = 5000;` be useful?